### ISSUE TITLE
[a11y]: Single date picker is not accessibility complaint

### ISSUE THREAD

**Issue by florenceluwork** [carbon-design-system/carbon#10150]
### Package

carbon-components-react

### Browser

Firefox

### Operating System

Windows

### Package version

7.44.0

### Automated testing tool and ruleset

N/A

### Assistive technology

JAWS

### Description

**Single date picker is not accessibility complaint:**
https://web-components.carbondesignsystem.com/?path=/docs/components-date-picker--default

I installed JAWS based on the instructions from this website: https://pages.github.ibm.com/IBMa/able/Test/screen-reader-tests/jaws-for-web/

- JAWS: 2021-2107.12 ILM
- FireFox: 94.0.1
- Windows 10 Enterprise
- 

**Issues:**
I was not able to use the Esc key to close the calendar picker once it is open
I was not able to use keyboard to navigate to the left / right / up / down arrow buttons. As a result, JAWS can't read the labels of these 4 buttons. 

### WCAG 2.1 Violation

https://www.w3.org/TR/WCAG21/#no-keyboard-trap

### CodeSandbox example

https://web-components.carbondesignsystem.com/?path=/docs/components-date-picker--default

### Steps to reproduce

**Issues:**
Open the Calendar Picker using Keyboard.
Hit the Esc key on the keyboard.
I was not able to use the Esc key to close the calendar picker once it is open.

Use the tab key, left, right, up or down arrow keys on the keyboard to navigate in the calendar picker. I was not able to use keyboard to navigate to the left / right / up / down arrow buttons. As a result, JAWS can't read the labels of these 4 buttons. 

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
Hi @florenceluwork, this looks like a bug report for the web components package. I'll transfer this to that repository.
